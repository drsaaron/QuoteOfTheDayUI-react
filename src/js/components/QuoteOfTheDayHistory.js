/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

import React, {Component} from 'react';
import QuoteOfTheDayHistoryForYear from './QuoteOfTheDayHistoryForYear';

export default class QuoteOfTheDayHistory extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        var years = Object.keys(this.props.history);
        var noHistory = years.length === 0 ? <em>No history</em> : "";
        return (
                <div id="quoteOfTheDayHistory">
                    <h2>Quote of the day history</h2>
                    {noHistory}
                    { years.map(year => <QuoteOfTheDayHistoryForYear key={year} year={year} history={this.props.history[year]} />)}
                </div>
                );
    }
}

