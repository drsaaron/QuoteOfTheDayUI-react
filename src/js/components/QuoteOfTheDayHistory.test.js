import React from 'react';
import renderer from 'react-test-renderer';

import QuoteOfTheDayHistory from './QuoteOfTheDayHistory';

describe('quote of the day history', () => {
    test('no history', () => {
	const history = {};
	const component = renderer.create(<QuoteOfTheDayHistory history={history} />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
    });

    test('with history', () => {
	const history = {
	    2001: [ { runDate: '2001-01-01' }, { runDate: '2001-02-01' } ],
	    2002: [ { runDate: '2002-10-01' } ]
	};
	const component = renderer.create(<QuoteOfTheDayHistory history={history} />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
    });
});
