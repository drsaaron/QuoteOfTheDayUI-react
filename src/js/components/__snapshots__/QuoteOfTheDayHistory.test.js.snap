// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quote of the day history no history 1`] = `
<div
  id="quoteOfTheDayHistory"
>
  <h2>
    Quote of the day history
  </h2>
  <em>
    No history
  </em>
</div>
`;

exports[`quote of the day history with history 1`] = `
<div
  id="quoteOfTheDayHistory"
>
  <h2>
    Quote of the day history
  </h2>
  <article>
    2001
    <ul>
      <li>
        2001-01-01
      </li>
      <li>
        2001-02-01
      </li>
    </ul>
  </article>
  <article>
    2002
    <ul>
      <li>
        2002-10-01
      </li>
    </ul>
  </article>
</div>
`;
